CMakeLists.txt

```diff
@@ -228,65 +265,26 @@ mark_as_advanced(QUDA_INTERFACE_NVTX)
 mark_as_advanced(QUDA_USE_EIGEN)
-mark_as_advanced(QUDA_BLOCKSOVER)
+mark_as_advanced(QUDA_BLOCKSOLVER)
 mark_as_advanced(QUDA_CXX_STANDARD)
 
 mark_as_advanced(QUDA_JITIFY)
 
 mark_as_advanced(QUDA_ARPACK_LOGGING)
 
-# ######################################################################################################################
-# everything below here is processing the setup
-# ######################################################################################################################
-# we need to check for some packages
-find_package(PythonInterp)
-
-# ######################################################################################################################
-# QUDA depends on Eigen this part makes sure we can download eigen if it is not found
-if(QUDA_DOWNLOAD_EIGEN)
-  set(EIGEN_VERSION 3.3.7)
+# some checks for invalid combinations
 
-  set(EIGEN_DOWNLOAD_LOCATION ${CMAKE_SOURCE_DIR}/externals/eigen/${EIGEN_VERSION}.tar.bz2)
-  set(EIGEN_URL http://bitbucket.org/eigen/eigen/get/${EIGEN_VERSION}.tar.bz2)
-  set(EIGEN_SHA 9f13cf90dedbe3e52a19f43000d71fdf72e986beb9a5436dddcd61ff9d77a3ce)
-  if(NOT EXISTS ${EIGEN_DOWNLOAD_LOCATION})
-    message(STATUS "Checking for Eigen tarball and downloading if necessary.")
-  endif()
-  file(DOWNLOAD ${EIGEN_URL} ${EIGEN_DOWNLOAD_LOCATION} EXPECTED_HASH SHA256=${EIGEN_SHA} STATUS EIGEN_DOWNLOADED)
-  list(GET EIGEN_DOWNLOADED 0 EIGEN_DOWNLOADED_CODE)
-  list(GET EIGEN_DOWNLOADED 1 EIGEN_DOWNLOADED_MSG)
-  if(${EIGEN_DOWNLOADED_CODE})
-    message(
-      SEND_ERROR
-        "Could not download Eigen automatically (${EIGEN_DOWNLOADED_MSG}). Please download eigen from ${EIGEN_URL} and save it to ${EIGEN_DOWNLOAD_LOCATION} and try running cmake again."
-      )
-  endif()
+if(QUDA_TEX AND QUDA_FLOAT8)
+  message(FATAL_ERROR "Cannot simultanteously enable QUDA_TEX and QUDA_FLOAT8")
+endif()
 
-  include(ExternalProject)
-  ExternalProject_Add(Eigen
-                      URL ${CMAKE_SOURCE_DIR}/externals/eigen/${EIGEN_VERSION}.tar.bz2
-                      URL_HASH SHA256=${EIGEN_SHA}
-                      PREFIX ${CMAKE_CURRENT_BINARY_DIR}/externals/eigen/
-                      CONFIGURE_COMMAND ""
-                      BUILD_COMMAND
-                      COMMAND ""
-                      INSTALL_COMMAND "")
-  ExternalProject_Get_Property(Eigen source_dir)
-  set(EIGEN_INCLUDE_DIRS ${source_dir})
-else()
-  # fall back to using find_package
-  find_package(Eigen QUIET)
-  if(NOT EIGEN_FOUND)
-    message(
-      FATAL_ERROR
-        "QUDA requires Eigen (http://eigen.tuxfamily.org). Please either set EIGEN_INCLUDE_DIRS to path to eigen3 include directory, e.g. /usr/local/include/eigen3 or set QUDA_DOWNLOAD_EIGEN to ON to enable automatic download of the necessary components."
-      )
-  endif()
+if(QUDA_MPI AND QUDA_QMP)
+  message(
+    WARNING
+      "Specifying QUDA_QMP and QUDA_MPI might result in undefined behavior. If you intend to use QMP set QUDA_MPI=OFF.")
```

Comment:
Should we just make this an error? Is there ever a reason why someone would want both QMP and MPI defined?

Reply:
Good idea. In fact if you enable both compilation will fail so better to error out.